nshlib/nsh_parse.c

```diff
@@ -2685,45 +2750,118 @@ static int nsh_parse_command(FAR struct nsh_vtbl_s *vtbl, FAR char *cmdline)
               argv[argc] = pbegin;
             }
         }
-    }
-
-  /* Check if the command should run in background */
+      else if (!strncmp(argv[argc], g_redirect_out2,
+                        strlen(g_redirect_out2)))
```

Comment:
should we save the result to avoid the dup at line 2759

Reply:
you mean, keep the value `2` in a `static const`, like `g_redirect_out2_len` ?

Reply:
```
len =  strlen(g_redirect_out2);
```
and use len at line 2734/2737/2739

Reply:
Sorry. I don't see how this is easily possible, as the first use is inside the condition to `else if` and the two others are inside the `{}` block. Where would this `len` be declared?

Reply:
could declared at the begin of function

Reply:
Done.